scripts/__python_argcomplete_run_windows

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
```

Comment:
I'm not comfortable with installing this script standalone on non-Windows platforms. Can we use setuptools conditionals to only install it on Windows?

Reply:
As far as I'm aware, there's nothing built in to setuptools that would let you conditionally include/exclude scripts (though I'm happy to be corrected). You could conditionally modify the thing you're passing to scripts in setup.py, but that would only work when installing from an sdist; for a wheel, which pip prefers, the decision is made at build time, meaning the script would be excluded globally if built on Linux and included globally if built on Windows. If we were to do that, we would have to update the instructions for Windows users telling them to `pip install argcomplete --no-binary argcomplete` so that they actually get the helper script.

I didn't take the approach of including the script under the argcomplete package and invoking it with `python -m argcomplete.__python_argcomplete_run_windows` for a couple of reasons. Firstly I don't necessarily know which interpreter I should be using to run the script, and secondly argcomplete does the majority of its work in its `__init__.py`, meaning running it that way would needlessly import all of argcomplete's dependencies, which negatively impacts response time.

So it seems our options are:
* Include the script unconditionally (it is useless on non-Windows platforms but also harmless; it will exit with an ImportError when it attempts to import msvcrt)
* Conditionally include the script in setup.py, advise Windows users to install from sdist
* Include only the handle writing changes, maintain the wrapper script in a separately installable package

I know this has been discussed several times before, but I'd like to take a moment to look back at the [tempfile IPC approach](https://github.com/evanunderscore/argcomplete/commit/481308a48efde8c8afd9aca4fef73033400e8d4a), which does have quite a few advantages:
* Does not require a helper script, so this packaging issue goes away
* Runs faster, since you don't need to launch an extra Python interpreter
* Does not need to guess using py.exe, since the shell launches the child directly
* No Windows-specific code is required, meaning this could be tested by Travis

Ultimately the decision is yours, so let me know how you would like to proceed with this.

(Irrespective of your decision here, I still believe directly passing a handle is the best option for PowerShell, since it has the capacity to create a pipe and provide the handle, whereas Git Bash does not.)

Reply:
Thanks for analyzing the trade-offs clearly.

I agree that the tempfile IPC approach is preferable in light of this. We should be able to activate tempfile IPC only on Windows, so that gets rid of the issue here.

Does this sound reasonable?

Reply:
(The main concern to watch out for with tempfiles is privacy issues from tempfiles being left behind; but I think given everything we discussed and given the scope limitation to Windows only, we should be able to mitigate that)

Reply:
I was intending to make temp files explicitly opt-in via an environment variable, but I can have it work by default on Windows if you'd prefer a more out of the box solution.

Creating the temporary files using `mktemp` should mean the files are only readable by the user (and a superuser) on both Windows and Linux, which means this won't be any less secure than your history file.

I'll prepare the other branch as a PR and we can continue the discussion there.